### PR TITLE
[gui/create-item] add whole body part spawning

### PR DESCRIPTION
**Pull request by ppaawwll**
add whole body part spawning (unbutchered, every layer) to create-item. changed creature material and body tissue layer selection prompt entries to be lowercase instead of uppercase, for ease of searching.

### CODE REVIEW

**Comment by ppaawwll** [comment]
🐇

File: changelog.txt [hunk 1]
```diff
@@ -29,14 +29,16 @@ that repo.
 - `makeown`: adds the selected unit as a member of your fortress
 
 ## Fixes
--@ `gui/unit-syndromes`: allow the window widgets to be interacted with
--@ `fix/protect-nicks`: now works by setting the historical figure nickname
--@ `gui/liquids`: fixed issues with unit pathing after adding/removing liquids
+- `gui/unit-syndromes`: allow the window widgets to be interacted with
+- `fix/protect-nicks`: now works by setting the historical figure nickname
+- `gui/liquids`: fixed issues with unit pathing after adding/removing liquids
+- `gui/create-item`: adding corpse spawning broke spawning generic corpsepieces, but has now been fixed.
```
Comment (myk002) [inline changelog.txt:35]: is this a merge error? these should all be prefixed with `-@`

File: gui/create-item.lua [hunk 2]
```diff
@@ -36,10 +36,10 @@ function getCreaturePartList(creatureID, casteID)
 end
 
 function getCreaturePartLayerList(creatureID, casteID, partID)
- local crplList={}
+ local crplList={{"whole"}}
  for k,crpl in ipairs(df.global.world.raws.creatures.all[creatureID].caste[casteID].body_info.body_parts[partID].layers) do
   local str = crpl.layer_name
-  table.insert(crplList,{str})
+  table.insert(crplList,{string.lower(str)}) -- lowercase string for easier searching
```
Comment (myk002) [inline gui/create-item.lua:42]: instead of making these lower case, you could change the FilteredList to `case_sensitive=false`

File: gui/create-item.lua [hunk 3]
```diff
@@ -224,16 +224,21 @@ function createCorpsePiece(creator, bodypart, partlayer, creatureID, casteID, ge
  creatureID = tonumber(creatureID)
  -- get the actual raws of the target creature
  local creatorRaceRaw = df.creature_raw.find(creatureID)
+ local wholePart = false
  casteID = tonumber(casteID)
  bodypart = tonumber(bodypart)
  partlayer = tonumber(partlayer)
+ if partlayer == -1 then -- somewhat similar to the bodypart variable below, a value of -1 here means that the user wants to spawn a whole body part. we set the partlayer to 0 (outermost) becuase the specific layer isn't important, and we're spawning them all anyway.
```
Comment (myk002) [inline gui/create-item.lua:231]:
```suggestion
 if partlayer == -1 then -- somewhat similar to the bodypart variable below, a value of -1 here means that the user wants to spawn a whole body part. we set the partlayer to 0 (outermost) because the specific layer isn't important, and we're spawning them all anyway.
```

File: gui/create-item.lua [hunk 4]
```diff
@@ -385,14 +391,20 @@ function createCorpsePiece(creator, bodypart, partlayer, creatureID, casteID, ge
        -- search through the target creature's body parts and bring back every one which has the desired material
        if creatorRaceRaw.tissue[creatorBody.body_parts[i].layers[n].tissue_id].tissue_material_str[1] == layerMat and creatorBody.body_parts[i].token ~= "SKULL" and not creatorBody.body_parts[i].flags.SMALL then
         item.body.components.body_part_status[i].missing = false
-        item.body.components.body_part_status[i].grime = 2
```
Comment (myk002) [inline gui/create-item.lua:388]: was this removal intentional? setting `grime` to 2 looks strange, but why was it here to begin with?

**Comment by myk002** [comment]
ping on this

File: gui/create-item.lua (hunk 4 above)
Comment (ppaawwll) [reply gui/create-item.lua:388]: it was. this was a leftover from me experimenting with what made corpsepieces work.

File: gui/create-item.lua (hunk 2 above)
Comment (ppaawwll) [reply gui/create-item.lua:42]: didn't know that, huh.

File: changelog.txt (hunk 1 above)
Comment (ppaawwll) [reply changelog.txt:35]: oops i did that manually and didn't know the format

**Comment by ppaawwll** [comment]
whoops

File: gui/create-item.lua (hunk 2 above)
Comment (ppaawwll) [reply gui/create-item.lua]: actually how do i do that? just stick it in?

**Comment by myk002** [comment]
did you mean to close this PR? it hasn't been merged yet.

**Comment by ppaawwll** [comment]
i updated my branch to the latest, which involves just wiping the commits it seems. opening another one in a sec.

**Comment by myk002** [comment]
did you lose any code? I might have a fetch of of your fork

**Comment by myk002** [comment]
resubmitted as #631
